packages/frontend/src/components/EditorSettings/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { ElementType, ReactNode, useMemo, useState } from 'react'
+import { BiMailSend } from 'react-icons/bi'
+import { Navigate, useParams } from 'react-router-dom'
+import {
+  Box,
+  Divider,
+  Flex,
+  useBreakpointValue,
+  VStack,
+} from '@chakra-ui/react'
+import * as URLS from 'config/urls'
+import useAuthentication from 'hooks/useAuthentication'
+
+import EditorDrawer from './EditorDrawer'
+import EditorSidebar from './EditorSidebar'
+import Navbar from './Navbar'
+
+export type DrawerLink = {
+  Icon: ElementType
+  text: string
+  to: string
+}
+
+export interface EditorSettingsLayoutProps {
+  children: ReactNode
+}
+
+export default function EditorSettingsLayout(
+  props: EditorSettingsLayoutProps,
+): React.ReactElement {
+  const { children } = props
+
+  const { currentUser } = useAuthentication()
+
+  const { flowId } = useParams()
+  const [isDrawerOpen, setDrawerOpen] = useState(false)
+
+  const openDrawer = () => setDrawerOpen(true)
+  const closeDrawer = () => setDrawerOpen(false)
+
+  const drawerLinks = useMemo(
+    () => [
+      {
+        Icon: BiMailSend,
+        text: 'Email notifications',
+        to: URLS.FLOW_EDITOR_NOTIFICATIONS(flowId),
+      },
+    ],
+    [flowId],
+  )
+
+  const drawerComponent = useBreakpointValue({
+    base: (
+      <>
+        <EditorDrawer
+          links={drawerLinks}
+          isDrawerOpen={isDrawerOpen}
+          openDrawer={openDrawer}
+          closeDrawer={closeDrawer}
+        />
+        <Divider borderColor="base.divider.medium" />
+      </>
+    ),
+    md: (
+      <>
+        <Box mt={4}>
+          <EditorSidebar links={drawerLinks} closeDrawer={closeDrawer} />
+        </Box>
+        <Divider orientation="vertical" borderColor="base.divider.medium" />
+      </>
+    ),
+  })
+
+  if (!currentUser) {
```

Comment:
comment: nice! out of scope but we can consider refactoring this into our `Page` / `LoginPage` component or simiar. Feels like "auto-redirect to last visited page after login" should be a common product-wide function.

Reply:
i refactored it out as a helper function instead, let's leave this to another PR to see if we can refactor all the redirects to `LoginPage`